lib/logstash/codecs/identity_map_codec.rb

```diff
@@ -182,11 +202,28 @@ def flush(&block)
 
   def close()
     cleaner.stop
+    auto_flusher.stop
     all_codecs.each(&:close)
   end
   # end Codec API
   # ==============================================
 
+  def auto_flush_mapped
+    nowf = Time.now.to_f
+    identity_map.each do |identity, compo|
+      next unless nowf > compo.auto_flush_timeout
+      compo.codec.auto_flush
```

Comment:
Yes, in my own test, I found that Time.now faster than `monotonic_time` and I am not really sure why.

I am okay to use float/integer since its faster.